### DISCUSSION TITLE
Working with multiple collections

### DISCUSSION THREAD

**Discussion by pietru2004** [pocketbase/pocketbase#5900] [Q&A]
Just to start I am bit confused and probably am missing something.
I have Collections called messages, userGroups, groups, and channel.
is it possible to via api rules allow listing messages by user only if one of groups they are in contains some kind of value that in name might contain channel id that is in json?
or should I use pb hooks?

I know in filters I can do to start @request.auth.id != "" && @collection.userGroups...
However I am not sure how to get 1 specific user and check in user groups via filters
User is relation to users (single), groups is relation to groups (multiple)

**Comment by ganigeorgiev** [comment]
The description is too vague and confusing (_"contains some kind of value that in name might contain channel id that is in json"_ - this doesn't really make sense to me).

Please try to be more specific by exporting your collections configuration as json (Admin UI > Settings > Export collections) and elaborate more clearly on what actually is the desired behavior.

**Reply by pietru2004** [reply]
so I am trying to make chat app with channels,
when user wants to see msgs from channel I want to allow it only if user has permission for that msgs.

userGroups - 1 user to many groups
group has permission json, where it would be contain array with something like {channel_id:id,read:true/false,...}

**Reply by pietru2004** [reply]
[pb_schema.json](https://github.com/user-attachments/files/17827647/pb_schema.json)

**Reply by ganigeorgiev** [reply]
No, sorry. If users refuse to clearly express and provide the required details then I'm not willing to help you. I consider it lazy and low-effort and I no longer want to encourage this type of behavior.

**Reply by pietru2004** [reply]
I am just not sure how to describe it

**Reply by ganigeorgiev** [reply]
No one can read your mind, nor has context about your project and what is the end goal.

You are simply not clearly explaining your requirements, what is the purpose of the channels' permissions, how they differ from the groups' permissions, which one need to be checked, etc.

As a general recommendation, avoid excessive usage of `@collection.*` fields.
If you want to access a field from the current authenticated user's group, then you can use something like: `@request.auth.userGroups_via_user.groups.permissions.something ?= "test"`.

If you need further help, try to elaborate more clearly what you are trying to do (if you are not sure whether your explanation is clear, you can always try to imagine yourself on the other side as a stranger that doesn't know anything about your project).

**Reply by pietru2004** [reply]
made new discussion where I hope I made clear and not complex https://github.com/pocketbase/pocketbase/discussions/5901
